zerver/tests/test_home.py

```diff
@@ -807,12 +815,12 @@ def test_generate_204(self) -> None:
         result = self.client_get("/api/v1/generate_204")
         self.assertEqual(result.status_code, 204)
 
-    def test_message_sent_time(self) -> None:
+    def test_user_activity_time(self) -> None:
         epoch_seconds = 1490472096
-        date_sent = datetime.datetime.fromtimestamp(epoch_seconds)
-        user_message = MagicMock()
-        user_message.message.date_sent = date_sent
-        self.assertEqual(sent_time_in_epoch_seconds(user_message), epoch_seconds)
+        activity_date = datetime.datetime.fromtimestamp(epoch_seconds)
+        user_activity = MagicMock()
+        user_activity.last_visit = activity_date
+        self.assertEqual(activity_time_in_epoch_seconds(user_activity), epoch_seconds)
```

Comment:
I'd like this test to instead just have another user send a message, have our user mark it as read, and then use an actual UserActivity object.  

It might also be worth doing a bit of refactoring along the lines of https://github.com/zulip/zulip/pull/12728 to pull the whole logic block for `furthest_read_time` we're testing here out into its own function?  That'd let us avoid calling `home_real` for the other unit test.